ext/AbstractFFTsChainRulesCoreExt.jl

```diff
@@ -159,4 +159,54 @@ function ChainRulesCore.rrule(::typeof(ifftshift), x::AbstractArray, dims)
     return y, ifftshift_pullback
 end
 
+# plans
+function ChainRulesCore.frule((_, _, Δx), ::typeof(*), P::AbstractFFTs.Plan, x::AbstractArray) 
+    y = P * x
+    if Base.mightalias(y, x)
+        throw(ArgumentError("differentiation rules are not supported for in-place plans"))
```

Comment:
Could you add tests that this error (and the others below) are thrown?

Reply:
That's a bit more involved if we actually want to probe it with FFT plans since currently the test suite does not contain any in-place test plans. I guess the easiest option would be to just re-use the existing out-of-place plans and define in-place updates re-using their implementations.

Reply:
I added tests :slightly_smiling_face: